### DISCUSSION TITLE
Allow JSON file uploads

### DISCUSSION THREAD

**Discussion by asportnoy** [community/community#42494] [Projects and Issues]
### Select Topic Area

Product Feedback

### Body

GitHub restricts file types that can be uploaded to issues/PRs/comments. However, the list of file types does not include JSON files. There have been a number of cases where I wanted to upload a JSON file but had to either convert it to a text file or create a separate gist for it. Both of those workarounds are not ideal for me and not ideal for whoever is receiving the file. Please add JSON to the allowed file types. Similar things like YAML files should also be allowed for the same reason as listed above.

**Comment by DaveSanders** [comment]
Yeah, I'm very confused by this limitation from github. Especially since they only seem to care about the file extension and not the contents. I'm wondering if its a security limitation?

**Reply by asportnoy** [reply]
My guess is security but JSON should be allowed.

**Comment by knipknap** [comment]
It's weird that this depends on file extensions anyway. I don't see how that raises security in a meaningful way. Better to use the MIME type...
